### PR TITLE
[mds-stream] Switch from NATS Streaming -> Vanilla NATS

### PR DESCRIPTION
**Pull request by avatarneil**
## PR Checklist

 - [ ] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [ ] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [ ] Provider
- [ ] Agency
- [ ] Audit
- [ ] Policy
- [ ] Compliance
- [ ] Daily
- [ ] Native
- [ ] Policy Author

### CODE REVIEW

**Comment by avatarneil** [comment]
This is in draft form right now because I'm waiting for the new `getEnvVar` method to land in `develop`, this will probably require a few tweaks after merging it in.

**Comment by avatarneil** [comment]
Note -- this PR is for the _client_ code to switch over.

File: packages/mds-stream/index.ts [hunk 1]
```diff
@@ -54,9 +53,7 @@ const getBinding = () => {
 
 const getNats = () => {
   if (!nats) {
-    nats = stan.connect(env.STAN_CLUSTER || 'stan', `mds-agency-${uuid()}`, {
-      url: `nats://${env.NATS}:4222`,
-      userCreds: env.STAN_CREDS,
+    nats = NATS.connect(`nats://${env.NATS}:4222`, {
```
Comment (drtyh2o) [inline packages/mds-stream/index.ts:56]: Shouldn't the entire URL (or `NATS_HOST`) come from `process.env` (perhaps using getEnvVar?

File: packages/mds-stream/nats-streaming/nats.ts [hunk 2]
```diff
@@ -8,7 +7,7 @@ export type EventProcessor<TData, TResult> = (type: string, data: TData) => Prom
 const SUBSCRIPTION_TYPES = ['event', 'telemetry'] as const
 type SUBSCRIPTION_TYPE = typeof SUBSCRIPTION_TYPES[number]
 
-const subscriptionCb = async <TData, TResult>(processor: EventProcessor<TData, TResult>, msg: stan.Message) => {
+const subscriptionCb = async <TData, TResult>(processor: EventProcessor<TData, TResult>, msg: any) => {
```
Comment (drtyh2o) [inline packages/mds-stream/nats-streaming/nats.ts:10]: No type for `msg` parameter?

Comment (avatarneil) [reply packages/mds-stream/nats-streaming/nats.ts:10]: Not yet (hoping to add it as part of this PR) -- I need to actually test the client still. It doesn't seem like NATS.js providers a type for the message; while I think they're a string, gotta double check.

File: packages/mds-stream/index.ts (hunk 1 above)
Comment (avatarneil) [reply packages/mds-stream/index.ts:56]: In theory, that isn't how we've structured the host to date, though. Would require some changes for our helm charts that I don't think we should implement until https://github.com/lacuna-tech/mds-core/pull/185 lands

Comment (drtyh2o) [reply packages/mds-stream/index.ts:56]: I think a `NATS_HOST` that defaults to `localhost:4222` would be consistent with how we handle other network services.

Comment (avatarneil) [reply packages/mds-stream/index.ts]: Refactored to use the new getEnvVar pattern, but haven't switched to `NATS_HOST` which would include protocol prefix and port suffix yet. Will wait on that until #185 lands.

**Pull request merged**